repo.go

```diff
@@ -105,11 +105,29 @@ func (r *Repo) Query(q string) (*Results, error) {
 	return results, nil
 }
 
-// Construct performs a SPARQL HTTP request to the Repo, and returns the
+// Construct performs a SPARQL HTTP query request to the Repo, and returns the
 // result triples.
 func (r *Repo) Construct(q string) ([]rdf.Triple, error) {
+	resp, err := r.send("query", q)
+	if err != nil {
+		return nil, err
+	}
+	dec := rdf.NewTripleDecoder(resp.Body, rdf.Turtle)
+	return dec.DecodeAll()
+}
+
+// Update performs a SPARQL HTTP update request to the Repo, and returns the
```

Comment:
This docstring is not quite correct, it does not return any triples, only an error if there is one.

Reply:
Okay, I've removed that part.